src/NuGet.CommandLine/Commands/UpdateCommand.cs

```diff
@@ -228,20 +228,44 @@ private Task UpdatePackagesAsync(string packagesConfigPath, INuGetProjectContext
             var projectDirectory = Path.GetDirectoryName(project.ProjectFullPath);
             var packageManager = new NuGetPackageManager(sourceRepositoryProvider, Settings, packagesDirectory);
             var nugetProject = new MSBuildNuGetProject(project, packagesDirectory, projectDirectory);
+            var versionConstraints = Safe ? VersionConstraints.ExactMajor | VersionConstraints.ExactMinor : VersionConstraints.None;
```

Comment:
Why don't you do a 'var resolutionContext' and use that in the 2 calls to PreviewUpdatePackagesAsync. That will make the code even more readable